regions/_geometry/core.pyx

```diff
@@ -37,6 +37,18 @@ ctypedef struct intersections:
     point p2
 
 
+cdef double floor_sqrt(double x):
+    """
+    In some of the geometrical functions, we have to take the sqrt of a number
+    and we know that the number should be >= 0. However, in some cases the
+    value is e.g. -1e-10, but we want to treat it as zero, which is what
```

Comment:
The issue is that then we have to then choose a threshold which will be architecture dependent. I would prefer to leave this as a function that we should call only if we are sure the result should be >= 0 modulo numerical uncertainty.